src/clients/api.ts

```diff
@@ -0,0 +1,235 @@
+import * as t from "io-ts";
+
+// A basic response type that also include 401
+import {
+  ApiHeaderJson,
+  basicErrorResponseDecoder,
+  basicResponseDecoder,
+  BasicResponseType,
+  composeHeaderProducers,
+  composeResponseDecoders,
+  createFetchRequestForApi,
+  IGetApiRequestType,
+  IPostApiRequestType,
+  IResponseType,
+  RequestHeaderProducer,
+  ResponseDecoder
+} from "italia-ts-commons/lib/requests";
+import nodeFetch from "node-fetch";
+import { ExtendedProfile } from "../types/api/ExtendedProfile";
+import { LimitedProfile } from "../types/api/LimitedProfile";
+import { MessageResponseWithContent } from "../types/api/MessageResponseWithContent";
+import { Messages } from "../types/api/Messages";
+import { ServicePublic } from "../types/api/ServicePublic";
+import { Services } from "../types/api/Services";
+
+const OcpApimSubscriptionKey = "Ocp-Apim-Subscription-Key";
+type OcpApimSubscriptionKey = t.TypeOf<typeof OcpApimSubscriptionKey>;
+
+// ProfileLimitedOrExtended is oneOf [LimitedProfile, ExtendedProfile]
+const ProfileLimitedOrExtended = t.union([LimitedProfile, ExtendedProfile]);
+
+export type ProfileLimitedOrExtended = t.TypeOf<
+  typeof ProfileLimitedOrExtended
+>;
+
+export type BasicResponseTypeWith401<R> =
+  | BasicResponseType<R>
+  | IResponseType<401, Error>;
+
+// A basic response decoder that also include 401
+export function basicResponseDecoderWith401<R, O = R>(
+  type: t.Type<R, O>
+): ResponseDecoder<BasicResponseTypeWith401<R>> {
+  return composeResponseDecoders(
+    basicResponseDecoder(type),
+    basicErrorResponseDecoder(401)
+  );
+}
+
+export function SubscriptionKeyHeaderProducer<P>(
+  token: string
+): RequestHeaderProducer<P, OcpApimSubscriptionKey> {
+  return () => ({
+    [OcpApimSubscriptionKey]: token
+  });
+}
+
+export type GetProfileT = IGetApiRequestType<
+  {
+    readonly fiscalCode: string;
```

Comment:
you can actually use `FiscalCode` here, instead of `string`

Reply:
Fixed from [6f74868] on